vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/RenderingManager.java

```diff
@@ -0,0 +1,315 @@
+
+        Context thymeleafContext = new Context();
+        populateContextForSpace(thymeleafContext, spaceId, sequenceHistory);
+        // add attributes to context
+        String response = springTemplateEngine.process(SPACE_TEMPLATE_DOWNLOAD_API, thymeleafContext);
+        byte[] fileContent = response.getBytes();
+        storageEngineDownloads.storeFile(fileContent, spaceId+DOWNLOAD_FILE_EXTENSION,spaceFolderPath );
+    }
+
+    /**
+     * 
+     * Downloads given module and related slides into space folder path.
+     * @param module            the {@link IModule} object indicating the module
+     * @param space             the {@link ISpace} object 
+     * @param imagesFolderPath  the path of the folder where the module's images will be stored
+     * @param spaceFolderPath   he path of the folder where the module's content will be stored
+     * 
+     */
+    private void downloadModule(IModule module, ISpace space, String imagesFolderPath, String spaceFolderPath) {
+        ISequence startSequence = module.getStartSequence();
+        if(startSequence!= null) {
+            try {
+                downloadSequences(startSequence, module, space, spaceFolderPath,imagesFolderPath );
+            } catch (FileStorageException e) {
+                logger.error("Could not download Module",e);
+            }
+        }
+    }
+       
+    /**
+     * 
+     * Downloads given sequence into spacefolderPath.
+     * @param startSequence      the {@link ISequence} object indicating the start sequence of the module
+     * @param module             the {@link IModule} object
+     * @param space              the {@link ISpace} object
+     * @param spaceFolderPath    the space folder path where space content will be stored
+     * @param imagesFolderPath   the images folder path where images will be stored
+     * @throws FileStorageException
+     * 
+     */
+    private void downloadSequences(ISequence startSequence, IModule module, ISpace space, String spaceFolderPath,
+            String imagesFolderPath) throws FileStorageException {
+        List<ISlide> slides = startSequence.getSlides();
+        slides.forEach(slide -> {
+            if(slide instanceof BranchingPoint) {
+                BranchingPoint branchingPoint = (BranchingPoint) slide;
+                List<IChoice> choices  =  branchingPoint.getChoices();
+                choices.forEach(choice -> {
+
+                    if(!choice.getSequence().getId().equals(startSequence.getId())) {
+
+                        try {
+                            downloadSequences(choice.getSequence(), module, space, spaceFolderPath, imagesFolderPath);
+                        } catch (FileStorageException e) {
+                            logger.error("Could not download Sequence",e);
+                        } 
+                    }
+                });
```

Comment:
a branching point is also a slide, it doesn't look like that page is ever rendered? or am I missing something?

Reply:
it was rendered before,  changed it to include branching point slide as well